pkg/agent/openflow/network_policy.go

```diff
@@ -1040,15 +1030,35 @@ func (c *client) UninstallPolicyRuleFlows(ruleID uint32) ([]string, error) {
 		return nil, err
 	}
 
-	for _, p := range ofPrioritiesToUninstallFlows {
-		conjsStalePriority, _ := c.policyCache.ByIndex(priorityIndex, p)
-		if len(conjsStalePriority) == 0 {
+	c.policyCache.Delete(conj)
+	return staleOFPriorities, nil
+}
+
+// getStalePriorities returns the ofPriorities that will be stale on the rule table where the
+// policyRuleConjunction is installed, after the deletion of that policyRuleConjunction.
+func (c *client) getStalePriorities(conj *policyRuleConjunction) (staleOFPriorities []string) {
+	var ofPrioritiesPotentiallyStale []string
+	if conj.ruleTableID != IngressRuleTable && conj.ruleTableID != EgressRuleTable {
+		ofPrioritiesPotentiallyStale = conj.ActionFlowPriorities()
```

Comment:
I do not understand why we concert priorities to string here. Could you check we can avoid such conversions to save CPU cycles?

Reply:
Because ActionFlowPriorities() is used as a indexer function and the interface requires []string as return type

Reply:
Are you saying indexer must take []string, or we can change it to use integers?

Reply:
It appears to only take []string
```
type Indexer interface {
        ....
	ByIndex(indexName, indexedValue string) ([]interface{}, error)
	// GetIndexer return the indexers
	GetIndexers() Indexers
	// AddIndexers adds more indexers to this store.  If you call this after you already have data
	// in the store, the results are undefined.
	AddIndexers(newIndexers Indexers) error
}

// IndexFunc knows how to compute the set of indexed values for an object.
type IndexFunc func(obj interface{}) ([]string, error)

// Indexers maps a name to a IndexFunc
type Indexers map[string]IndexFunc
```

Reply:
correct.. indexers only take []string

Reply:
ok